test/ATokenCheckpointed.t.sol

```diff
@@ -0,0 +1,1974 @@
+
+    // Express voting preferences.
+    if (aToken.balanceOf(_userA) == 0) vm.expectRevert(bytes("no weight"));
+    vm.prank(_userA);
+    aToken.expressVote(_proposalId, _supportTypeA);
+    vm.prank(_userB);
+    aToken.expressVote(_proposalId, _supportTypeB);
+
+    // Wait until after the pool's voting period closes.
+    vm.roll(aToken.internalVotingPeriodEnd(_proposalId) + 1);
+
+    // Submit votes on behalf of the pool.
+    aToken.castVote(_proposalId);
+
+    (uint256 _againstVotes, uint256 _forVotes, uint256 _abstainVotes) =
+      governor.proposalVotes(_proposalId);
+
+    if (_supportTypeA == _supportTypeB) {
+      if (_supportTypeA == uint8(VoteType.For)) assertEq(_forVotes, _weight);
+      if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weight);
+      if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weight);
+    } else {
+      // forgefmt: disable-start
+      if (_supportTypeA == uint8(VoteType.For)) assertEq(_forVotes, _weight - _transferAmount);
+      if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weight - _transferAmount);
+      if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weight - _transferAmount);
+      if (_supportTypeB == uint8(VoteType.For)) assertEq(_forVotes, _transferAmount);
+      if (_supportTypeB == uint8(VoteType.Against)) assertEq(_againstVotes, _transferAmount);
+      if (_supportTypeB == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _transferAmount);
+      // forgefmt: disable-end
+    }
+  }
+}
+
+contract GetPastStoredBalanceTest is AaveAtokenForkTest {
+  function test_GetPastStoredBalanceCorrectlyReadsCheckpoints() public {
+    _initiateRebasing();
+
+    address _who = address(0xBEEF);
+    uint256 _amountA = 42 ether;
+    uint256 _amountB = 3 ether;
+
+    uint256[] memory _rawBalances = new uint256[](3);
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_who, _amountA);
+    _rawBalances[0] = aToken.exposedRawBalanceOf(_who);
+
+    // It's important that this be greater than a ray, since Aave uses this
+    // index when determining the raw stored balance. If it were a ray, the
+    // stored balance would just equal the supplied amount and this test would
+    // be less meaningful.
+    assertGt(
+      pool.getReserveData(address(govToken)).liquidityIndex,
+      1e27,
+      "liquidityIndex has not changed, is rebasing occuring?"
+    );
+
+    // The supplied amount should be less than the raw balance, which was
+    // scaled down by the reserve liquidity index.
+    assertLt(_rawBalances[0], _amountA, "supply wasn't reduced by liquidityIndex");
+
+    // Advance the clock.
+    uint256 _blocksJumped = 42;
+    vm.roll(block.number + _blocksJumped);
+    vm.warp(block.timestamp + 42 days);
+
+    // Rebasing should not affect the raw balance.
+    assertEq(
+      aToken.exposedRawBalanceOf(_who),
+      _rawBalances[0],
+      "raw balance shouldn't be affected by rebasing"
+    );
```

Comment:
This kind of feels like a no-op assertion. We read a storage slot on L1635, make no state changes (aside from skipping forward in blocks/time), and read the same storage slot again on L1658. Not really a problem I guess, just seems a bit odd since of course the slot won't change 🤷

Maybe it's really just validating that `exposedRawBalanceOf` is reading the right slot/value? Reading `getPastStoredBalance` does make more sense though, since that tests our implementation

Reply:
Yeah, I don't think we need it. I may have just had it in here while I was TDD-ing the feature

Reply:
Removed in aa6ca4c